plugins/modules/web_infrastructure/jenkins_build.py

```diff
@@ -0,0 +1,251 @@
+
+    def get_next_build(self):
+        try:
+            build_number = self.server.get_job_info(self.name)['nextBuildNumber']
+        except Exception as e:
+            self.module.fail_json(msg='Unable to get job info from Jenkins server, %s' % to_native(e), exception=traceback.format_exc())
+
+        return build_number
+
+    def get_build_status(self):
+        try:
+            response = self.server.get_build_info(self.name, self.build_number)
+            return response
+
+        except Exception as e:
+            self.module.fail_json(msg='Unable to fetch build information, %s' % to_native(e), exception=traceback.format_exc())
+
+    def present_build(self):
+        self.build_number = self.get_next_build()
+
+        try:
+            if self.args is None:
+                self.server.build_job(self.name)
+            else:
+                self.server.build_job(self.name, self.args)
+        except Exception as e:
+            self.module.fail_json(msg='Unable to create build for %s: %s' % (self.jenkins_url, to_native(e)),
+                                  exception=traceback.format_exc())
+
+    def absent_build(self):
+        try:
+            self.server.delete_build(self.name, self.build_number)
+        except Exception as e:
+            self.module.fail_json(msg='Unable to delete build for %s: %s' % (self.jenkins_url, to_native(e)),
+                                  exception=traceback.format_exc())
+
+    def get_result(self):
+        result = self.result
+        build_status = self.get_build_status()
+
+        if build_status['result'] is None:
+            sleep(10)
+            self.get_result()
+        else:
+            if build_status['result'] == "SUCCESS":
+                result['changed'] = True
+                result['build_info'] = build_status
+            else:
+                result['failed'] = True
+                result['build_info'] = build_status
+
+        return result
+
+
+def test_dependencies(module):
+    if not python_jenkins_installed:
+        module.fail_json(
+            msg=missing_required_lib("python-jenkins",
+                                     url="https://python-jenkins.readthedocs.io/en/latest/install.html"),
+            exception=JENKINS_IMP_ERR)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            args=dict(required=False, type='dict'),
+            build_number=dict(required=False, type='int'),
+            name=dict(required=True),
+            password=dict(required=False, no_log=True),
+            state=dict(required=False, choices=['present', 'absent'], default="present"),
+            token=dict(required=False, no_log=True),
+            url=dict(required=False, default="http://localhost:8080"),
+            user=dict(required=False)
+        ),
+        mutually_exclusive=[
+            ['password', 'token'],
+        ],
+        supports_check_mode=True,
```

Comment:
```suggestion
        supports_check_mode=True,
```
if this is really supports `check_mode` (i don't see it's implementation in the code), please put it in the doc, e.g.:
```
note:
  - Supports C(check_mode).
```